src/tracked_struct.rs

```diff
@@ -435,7 +431,11 @@ where
         };
 
         if let Some(id) = self.free_list.pop() {
-            Self::data_raw(zalsa.table(), id).with_mut(|data_raw| {
+            // Increment the ID generation before reusing it, as if we have allocated a new
+            // slot in the table.
+            let id = id.with_generation(id.generation() + 1);
```

Comment:
Nit: I think it would be nice to have a `id.increment_generation` or `new_generation`. We should also handle the case where generation overflows.